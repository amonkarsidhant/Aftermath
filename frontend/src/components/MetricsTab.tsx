import { LineChart, Line, XAxis, YAxis, Tooltip, ResponsiveContainer } from 'recharts';
import { mttrData } from '../utils/mock/metrics';

export default function MetricsTab() {
  return (
    <div className="w-full min-w-0 h-48 md:h-64 p-4 bg-white dark:bg-neutral-800 border border-neutral-200 dark:border-neutral-700 rounded">
      <ResponsiveContainer>
        <LineChart data={mttrData}>
          <XAxis dataKey="date" />
          <YAxis />
          <Tooltip />
          <Line type="monotone" dataKey="mttr" stroke="#8884d8" />
        </LineChart>
      </ResponsiveContainer>
    </div>
  );
}

