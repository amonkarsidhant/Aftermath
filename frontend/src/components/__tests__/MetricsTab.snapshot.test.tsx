import { render } from '@testing-library/react';
import MetricsTab from '../MetricsTab';

jest.mock('recharts', () => ({
  ResponsiveContainer: ({ children }: any) => <div>{children}</div>,
  LineChart: ({ children }: any) => <div>{children}</div>,
  Line: () => <div />,
  XAxis: () => <div />,
  YAxis: () => <div />,
  Tooltip: () => <div />,
}));

describe('MetricsTab snapshot', () => {
  it('matches snapshot', () => {
    const { container } = render(<MetricsTab />);
    expect(container).toMatchSnapshot();
  });
});
