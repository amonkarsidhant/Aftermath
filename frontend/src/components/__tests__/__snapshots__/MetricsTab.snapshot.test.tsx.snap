exports[`MetricsTab snapshot matches snapshot 1`] = `
<div class="w-full min-w-0 h-48 md:h-64 p-4 bg-white dark:bg-neutral-800 border border-neutral-200 dark:border-neutral-700 rounded">
  <div>
    <div>
      <div />
      <div />
      <div />
      <div />
    </div>
  </div>
</div>
`;
